src/caret_analyze/infra/lttng/ros2_tracing/data_model_service.py

```diff
@@ -127,8 +137,47 @@ def ns_and_node_name(row: pd.Series) -> str:
             else:
                 return ns + '/' + name
 
-        match_nodes = self._data.nodes.loc[node_handles, :]
+        try:
+            match_middle = self._ensure_dataframe(middle_df.loc[handle, :])
+            node_handles = match_middle.loc[:, 'node_handle'].to_list()
+            match_nodes = self._data.nodes.loc[node_handles, :]
+            assert len(match_nodes) == 1
+            return ns_and_node_name(match_nodes.iloc[0])
+        except KeyError:
+            return None
+
+    def _get_callback_symbol_from_handle(
+        self,
+        handle: int
+    ) -> Optional[str]:
+        try:
+            match_callback_objects = self._ensure_dataframe(
+                self._data.callback_objects.loc[handle, :])
+            callback_objects = match_callback_objects.loc[:, 'callback_object'].to_list()
+            match_callback_symbols = self._data.callback_symbols.loc[callback_objects, :]
+            assert len(match_callback_symbols) == 1
```

Comment:
Repeated creation and deletion of nodes may destroy this assumption

It would look something like this
```
    def _get_callback_symbol_from_handle(
        self,
        handle: int
    ) -> List[str]:
```

Reply:
@hsgwa 
Thank you very much for pointing that out.
Fixed in: https://github.com/tier4/CARET_analyze/pull/182/commits/16b84a41b9a77cecc381776e8226958ac5a64c60